src/services/profile-service.ts

```diff
@@ -0,0 +1,27 @@
+import { PostgrestError, SupabaseClient } from '@supabase/supabase-js';
+import { DbProfile } from '../types/aliases';
+import { Database } from '../types/database.types';
+
+type ValidRole = 'manager' | 'donor'
+
+export class ProfileService {
+  supabase: SupabaseClient<Database>;
+
+  constructor(supabase: SupabaseClient<Database>) {
+    this.supabase = supabase;
+  }
+
+  async fetchProfiles(roles: ValidRole[], onSuccess: (projects: DbProfile[]) => void, onError: (error: PostgrestError) => void) {
+
+    const { data, error } = await this.supabase.from('profile').select(`
```

Comment:
We can create a `view` for this, and then query that so that the code here is a bit simpler, but that can be done in a future PR if we care about it.

Maybe it isn't the worst thing for now to just have the queries here and then we can take a view as to what we might want to hide behind a view.